src/pages/RealtimeMapPage.tsx

```diff
@@ -60,66 +65,87 @@ function numberToColorHsl(i: number, max: number) {
-  const [limit, setLimit] = useState(500)
-  const [loaded, setLoaded] = useState(0)
   const [from, setFrom] = useState('2023-05-01T12:00:00+02:00')
   const [to, setTo] = useState('2023-05-01T12:01:00+02:00')
 
-  useEffect(() => {
-    ;(async () => {
-      const url = `https://open-bus-stride-api.hasadna.org.il/siri_vehicle_locations/list?get_count=false&recorded_at_time_from=${
-        formatTime(from) || '2023-05-01T12:00:00+02:00'
-      }&recorded_at_time_to=${
-        formatTime(to) || '2023-05-01T12:01:00+02:00'
-      }&order_by=id%20asc&limit=${limit}`
-      const response = await fetch(url)
-      const data = await response.json()
-      const points: Point[] = data.map((point: { lat: number; lon: number; velocity: number }) => ({
-        loc: [point.lat, point.lon],
-        color: point.velocity,
-        point,
-      }))
-      setPositions(points)
-      setLoaded(points.length)
-    })()
-  }, [limit])
+  const locations = useVehicleLocations({
+    from: formatTime(from),
+    to: formatTime(to),
+  })
+
+  const loaded = locations.length
+
+  const positions = useMemo(() => {
+    let pos = locations.map<Point>((location) => ({
+      loc: [location.lat, location.lon],
+      color: location.velocity,
+      operator: location.siri_route__operator_ref,
+      bearing: location.bearing,
+      recorded_at_time: new Date(location.recorded_at_time).getTime(),
+      point: location,
+    }))
+    // keep only the latest point for each vehicle
+    pos = pos.filter((p) =>
+      pos.every(
+        (p2) =>
+          p2.point!.siri_ride__vehicle_ref !== p.point!.siri_ride__vehicle_ref ||
+          p2.recorded_at_time! <= p.recorded_at_time!,
+      ),
+    )
+    return pos
+  }, [locations])
 
   return (
     <div className="map-container">
       <div className="map-header">
         <h1>Realtime Map</h1>
-        <div className="map-header-buttons">
-          <button onClick={() => setLimit(500)}>500</button>
-          <button onClick={() => setLimit(1000)}>1000</button>
-          <button onClick={() => setLimit(5000)}>5000</button>
-          <button onClick={() => setLimit(10000)}>10000</button>
-          <button onClick={() => setLimit(10000)}>30000</button>
-        </div>
         <div className="map-header-buttons">
           <label>
             מתאריך
-            <input type="datetime-local" value={from} onChange={(e) => setFrom(e.target.value)} />
+            <input
+              type="datetime-local"
+              value={from.slice(0, 16)}
+              onChange={(e) => {
+                setFrom(e.target.value)
+                setTo(formatTime(+new Date(e.target.value) + (+new Date(to) - +new Date(from))))
+              }}
+            />
           </label>{' '}
           {` `}
           <label>
-            עד תאריך
-            <input type="datetime-local" value={to} onChange={(e) => setTo(e.target.value)} />
+            צפה במיקומי אוטובוסים בטווח של
```

Comment:
I tried to keep all strings in one place (texts.ts or something) to make translation and compatibility easier.

Reply:
done